Build failure

With latest sources:
```
ninja: Entering directory `build'
[363/611] Compiling C object sway/sway.p/lock.c.o
FAILED: sway/sway.p/lock.c.o
cc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Isubprojects/wlroots/include -I../subprojects/wlroots/include -Iprotocols -I/usr/include/cairo -I/usr/include/lzo -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O0 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wundef -Wvla '-DSYSCONFDIR="//usr/local/etc"' '-DSWAY_VERSION="1.9-dev-3bd657c7 (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/lock.c.o -MF sway/sway.p/lock.c.o.d -o sway/sway.p/lock.c.o -c ../sway/lock.c
../sway/lock.c: In function ‘handle_new_surface’:
../sway/lock.c:119:50: error: ‘struct <anonymous>’ has no member named ‘mode’
  119 |         wl_signal_add(&output->wlr_output->events.mode, &surf->output_mode);
      |                                                  ^
[376/611] Compiling C object sway/sway.p/desktop_output.c.o
FAILED: sway/sway.p/desktop_output.c.o
cc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Isubprojects/wlroots/include -I../subprojects/wlroots/include -Iprotocols -I/usr/include/cairo -I/usr/include/lzo -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O0 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wundef -Wvla '-DSYSCONFDIR="//usr/local/etc"' '-DSWAY_VERSION="1.9-dev-3bd657c7 (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/desktop_output.c.o -MF sway/sway.p/desktop_output.c.o.d -o sway/sway.p/desktop_output.c.o -c ../sway/desktop/output.c
../sway/desktop/output.c: In function ‘handle_new_output’:
../sway/desktop/output.c:957:42: error: ‘struct <anonymous>’ has no member named ‘mode’
  957 |         wl_signal_add(&wlr_output->events.mode, &output->mode);
      |                                          ^
[380/611] Linking target subprojects/wlroots/libwlroots.so.12
ninja: build stopped: subcommand failed.
```

> Can you try #7305 ?

That is still failing for me:
```
ninja: Entering directory `build'
[379/611] Compiling C object sway/sway.p/desktop_output.c.o
FAILED: sway/sway.p/desktop_output.c.o
cc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Isubprojects/wlroots/include -I../subprojects/wlroots/include -Iprotocols -I/usr/include/cairo -I/usr/include/lzo -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O0 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wundef -Wvla '-DSYSCONFDIR="//usr/local/etc"' '-DSWAY_VERSION="1.9-dev-71d968bb (" __DATE__ ", branch '"'"'wlroots-remove-output-mode'"'"')"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/desktop_output.c.o -MF sway/sway.p/desktop_output.c.o.d -o sway/sway.p/desktop_output.c.o -c ../sway/desktop/output.c
../sway/desktop/output.c: In function ‘handle_commit’:
../sway/desktop/output.c:864:17: error: implicit declaration of function ‘output_mode’; did you mean ‘output_render’? [-Werror=implicit-function-declaration]
  864 |                 output_mode(output);
      |                 ^~~~~~~~~~~
      |                 output_render
../sway/desktop/output.c: At top level:
../sway/desktop/output.c:824:13: error: ‘handle_mode’ defined but not used [-Werror=unused-function]
  824 | static void handle_mode(struct sway_output *output) {
      |             ^~~~~~~~~~~
cc1: all warnings being treated as errors
[396/611] Compiling C object sway/sway.p/input_seat.c.o
ninja: build stopped: subcommand failed.
```

--- a/include/sway/output.h
+++ b/include/sway/output.h
@@ -43,7 +43,6 @@ struct sway_output {
 
 	struct wl_listener destroy;
 	struct wl_listener commit;
-	struct wl_listener mode;
 	struct wl_listener present;
 	struct wl_listener damage;
 	struct wl_listener frame;

--- a/sway/desktop/output.c
+++ b/sway/desktop/output.c
@@ -805,7 +805,6 @@ static void handle_destroy(struct wl_listener *listener, void *data) {
 
 	wl_list_remove(&output->destroy.link);
 	wl_list_remove(&output->commit.link);
-	wl_list_remove(&output->mode.link);
 	wl_list_remove(&output->present.link);
 	wl_list_remove(&output->damage.link);
 	wl_list_remove(&output->frame.link);
@@ -822,8 +821,7 @@ static void handle_destroy(struct wl_listener *listener, void *data) {
 	update_output_manager_config(server);
 }
 
-static void handle_mode(struct wl_listener *listener, void *data) {
-	struct sway_output *output = wl_container_of(listener, output, mode);
+static void handle_mode(struct sway_output *output) {
 	if (!output->enabled && !output->enabling) {
 		struct output_config *oc = find_output_config(output);
 		if (output->wlr_output->current_mode != NULL &&
@@ -861,6 +859,10 @@ static void update_textures(struct sway_container *con, void *data) {
 static void handle_commit(struct wl_listener *listener, void *data) {
 	struct sway_output *output = wl_container_of(listener, output, commit);
 	struct wlr_output_event_commit *event = data;
+
+	if (event->committed & WLR_OUTPUT_STATE_MODE) {
+		handle_mode(output);
+	}
 
 	if (!output->enabled) {
 		return;
@@ -954,8 +956,6 @@ void handle_new_output(struct wl_listener *listener, void *data) {
 	output->destroy.notify = handle_destroy;
 	wl_signal_add(&wlr_output->events.commit, &output->commit);
 	output->commit.notify = handle_commit;
-	wl_signal_add(&wlr_output->events.mode, &output->mode);
-	output->mode.notify = handle_mode;
 	wl_signal_add(&wlr_output->events.present, &output->present);
 	output->present.notify = handle_present;
 	wl_signal_add(&wlr_output->events.damage, &output->damage);

--- a/sway/lock.c
+++ b/sway/lock.c
@@ -13,7 +13,6 @@ struct sway_session_lock_surface {
 	struct wl_listener map;
 	struct wl_listener destroy;
 	struct wl_listener surface_commit;
-	struct wl_listener output_mode;
 	struct wl_listener output_commit;
 	struct wl_listener output_destroy;
 };
@@ -38,12 +37,6 @@ static void handle_surface_map(struct wl_listener *listener, void *data) {
 static void handle_surface_commit(struct wl_listener *listener, void *data) {
 	struct sway_session_lock_surface *surf = wl_container_of(listener, surf, surface_commit);
 	output_damage_surface(surf->output, 0, 0, surf->surface, false);
-}
-
-static void handle_output_mode(struct wl_listener *listener, void *data) {
-	struct sway_session_lock_surface *surf = wl_container_of(listener, surf, output_mode);
-	wlr_session_lock_surface_v1_configure(surf->lock_surface,
-		surf->output->width, surf->output->height);
 }
 
 static void handle_output_commit(struct wl_listener *listener, void *data) {
@@ -76,7 +69,6 @@ static void destroy_lock_surface(struct sway_session_lock_surface *surf) {
 	wl_list_remove(&surf->map.link);
 	wl_list_remove(&surf->destroy.link);
 	wl_list_remove(&surf->surface_commit.link);
-	wl_list_remove(&surf->output_mode.link);
 	wl_list_remove(&surf->output_commit.link);
 	wl_list_remove(&surf->output_destroy.link);
 	output_damage_whole(surf->output);
@@ -115,8 +107,6 @@ static void handle_new_surface(struct wl_listener *listener, void *data) {
 	wl_signal_add(&lock_surface->events.destroy, &surf->destroy);
 	surf->surface_commit.notify = handle_surface_commit;
 	wl_signal_add(&surf->surface->events.commit, &surf->surface_commit);
-	surf->output_mode.notify = handle_output_mode;
-	wl_signal_add(&output->wlr_output->events.mode, &surf->output_mode);
 	surf->output_commit.notify = handle_output_commit;
 	wl_signal_add(&output->wlr_output->events.commit, &surf->output_commit);
 	surf->output_destroy.notify = handle_output_destroy;

